File: sdk/typescript/examples/fastapi_app.ts
import { memoryOSApiKey, memoryOSBaseUrl } from "./env.js";
import { MemoryOS } from "../src/index.js";

const memoryClient = new MemoryOS(memoryOSApiKey(), memoryOSBaseUrl());

export async function assistantContext(query: string, externalUserId: string): Promise<{
  quotaMode: string;
  systemPromptAddition: string;
  contextTokenCount: number;
  memories: string[];
}> {
  const results = await memoryClient.get({
    query,
    externalUserId,
    limit: 5,
    contextMaxTokens: 300,
  });
  return {
    quotaMode: results.quotaMode,
    systemPromptAddition: results.hasContext ? results.systemPromptAddition : "",
    contextTokenCount: results.contextTokenCount,
    memories: results.items.map((item) => item.content),
  };
}

File: sdk/python/examples/fastapi_app.py
import os

from fastapi import FastAPI

from memoryos import AsyncMemory


app = FastAPI()
memory_client = AsyncMemory(
    api_key=os.environ["MEMORYOS_API_KEY"],
    base_url=os.getenv("MEMORYOS_BASE_URL", "http://127.0.0.1:8000"),
)


@app.get("/assistant-context")
async def assistant_context(query: str, external_user_id: str) -> dict[str, object]:
    results = await memory_client.get(
        query=query,
        external_user_id=external_user_id,
        limit=5,
        context_max_tokens=300,
    )
    return {
        "quota_mode": results.quota_mode,
        "system_prompt_addition": results.system_prompt_addition if results.has_context else "",
        "context_token_count": results.context_token_count,
        "memories": [item.content for item in results.items],
    }
